src/databricks_ai_bridge/vector_search_retriever_tool.py

```diff
@@ -62,6 +63,7 @@ class VectorSearchRetrieverToolMixin(BaseModel):
     resources: Optional[List[dict]] = Field(
         None, description="Resources required to log a model that uses this tool."
     )
+    credential_strategy: Optional[CredentialStrategy] = Field(default=None, description="When set to MODEL_SERVING_USER_CREDENTIALS, the tool will be authorized with on behalf of user rights")
```

Comment:
Overall i am curious if we should ask users to pass in the workspace_client instead? So that we can support things like profile, credential_strategy, etc. What do you think? This will be similar to other APIs as well which just cares about passing in the workspaceClient. So if the client is already created we don't need to pass in just the credential_strategy here. Thoughts?

Reply:
@smurching I think this is a good idea to have a WorkspaceClient being passed into both VectorSearchTools and UCFunctionToolkit. Internally we could take the workspace client and pass in the credential strategy when we are actually creating the VectorSearchClient since VectorSearchClient does not take in WorkspaceClient.

On the plus side, it would be a consistent CUJ for our users.

on the negative, if users are using the VectorSearchClient directly, then they would have to pass in this credential strategy explicitly. So using VectorSearchClient directly vs using a VectorSearchTool would have different parameters. Is that ok? What do you think

Reply:
What's the overall plan for the vector search SDK (it will eventually move to Databricks SDK right?). I think it's fine for us to be consistent with the vector search SDK for now (pass individual params like `credential_strategy` in both places) since to Aravind's point, it may also be confusing to pass a `WorkspaceClient` and find that only the subset of WorkspaceClient attributes that are supported by the VS client are honored in the tool. But definitely long term I'd hope we can pass a WorkspaceClient here

Reply:
But if there aren't a lot of config parameters, and the VS SDK is going to merge into Databricks SDK soon, then maybe we should just pass WorkspaceClient upfront

Reply:
Yeah! that is what i was thinking that in the long run anyway, VS SDK would be deprecated and migrated to workspaceClient. if that happens, there won't be any change in our API.
Plus for users we recommend using UCFunctionToolkit and databricks-ai-bridge for vector search, etc. For all those usecases we can expect workspaceClient then its super consistent for the users

Reply:
@smurching @sunishsheth2009 I am not sure if there is a target date for VS Client to migrate to Workspace Client, but I know its going to happen at some point. But if we want to steer our users to use the VS Tool from ai-bridge and not really promote the VectorSearchClient directly, then I think taking in WorkspaceClient here makes sense